### PR TITLE
[RFC] Refactor

### PR DESCRIPTION
**Pull request by jleveugle**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +0,0 @@
-* text=auto
```
Comment (antleblanc) [inline .gitattributes]: Please do not remove this file.

File: package.json [hunk 2]
```diff
@@ -5,29 +5,58 @@
...
   "repository": "ovh-ux/ovh-utils-angular",
+  "main": "./src/index.js",
   "scripts": {
-    "lint": "eslint src/. --ext .js --quiet",
-    "test": "grunt test"
+    "test": "yarn run lint",
+    "lint": "run-p lint:css lint:js",
+    "lint:css": "stylelint 'src/**/*.less' --fix",
+    "lint:js": "eslint --quiet --fix ./src",
+    "version": "npx conventional-changelog-cli -p angular -i CHANGELOG.md -s && git add CHANGELOG.md",
```
Comment (antleblanc) [inline package.json]: missing `"postversion"` hook I guess.

File: package.json [hunk 3]
```diff
@@ -5,29 +5,58 @@
...
+    "build": "rollup -c",
+    "prepublish": "rollup -c"
   },
   "keywords": [
     "angular",
     "utils"
   ],
+  "lint-staged": {
+    "*.less": [
```
Comment (antleblanc) [inline package.json]: missing *.js as well.

File: src/alerter/alerter-directive.js [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
-            }
+export default function () {
+  return {
+    restrict: 'A',
+    scope: {
+      ovhAlert: '@',
+    },
+    transclude: true,
+    template,
+    link($scope, $elm, $attr) {
```
Comment (antleblanc) [inline src/alerter/alerter-directive.js]: ngInject?

File: src/alerter/alerter-service.js [hunk 5]
```diff
@@ -1,109 +1,110 @@
...
+
+    let alertType = '';
+    if (data) {
+      if (data.message) {
+        messageToSend += ` (${data.message})`;
+        alertType = alertTypesHash[data.type];
+      } else if (data.messages) {
+        if (data.messages.length > 0) {
+          alertType = alertTypesHash[data.state];
+          messagesFiltered = $.grep(data.messages, e => e.type && e.type !== 'INFO');
```
Comment (antleblanc) [inline src/alerter/alerter-service.js]: Does `$.grep` require an `import` statement?

File: src/alerter/alerter.js [hunk 6]
```diff
@@ -1 +1,16 @@
...
+import alerterService from './alerter-service';
+
+const moduleName = 'ua.alerter';
+
+angular
+  .module(moduleName, [tooltipBox, translate])
+  .service('Alerter', alerterService)
+  .directive('ovhAlert', alerterDirective);
+
+export default moduleName;
```
Comment (antleblanc) [inline src/alerter/alerter.js]: Is there any difference with:

```js
export default angular
  .module('ua.alerter', […])
  .…
  .name;
```

File: src/contracts/contracts-directive.js [hunk 7]
```diff
@@ -1,121 +1,139 @@
...
+  return {
+    restrict: 'EA',
+    replace: true,
+    require: '^ngModel',
+    template,
+    scope: {
+      contracts: '=',
+      agree: '=ngModel',
+    },
+    link($scope, $elm, $attr) {
```
Comment (antleblanc) [inline src/contracts/contracts-directive.js]: missing `ngInject`?

File: src/alerter/alerter-directive.js (hunk 4 above)
Comment (cbourgois) [reply src/alerter/alerter-directive.js]: Nope, no injection in `link`

File: package.json (hunk 2 above)
Comment (jleveugle) [reply package.json]: done

File: package.json (hunk 3 above)
Comment (jleveugle) [reply package.json]: done

File: src/alerter/alerter-service.js (hunk 5 above)
Comment (jleveugle) [reply src/alerter/alerter-service.js]: + env jQuery

File: src/contracts/contracts-directive.js (hunk 7 above)
Comment (jleveugle) [reply src/contracts/contracts-directive.js]: Nop 

https://docs.angularjs.org/api/ng/service/$compile#-link-

Link is called by angular itself :)

**Pull request closed without merging**